records/generator.go

```diff
@@ -529,58 +575,30 @@ func (rg *RecordGenerator) setFromLocal(host string, ns string) {
 				continue
 			}
 
-			rg.insertRR(ns, ip.String(), "A")
+			rg.insertRR(ns, ip.String(), A)
 		}
 	}
 }
 
-func (rg *RecordGenerator) exists(name, host, rtype string) bool {
-	if rtype == "A" {
-		if val, ok := rg.As[name]; ok {
-			// check if A record already exists
-			// identical tasks on same slave
-			for _, b := range val {
-				if b == host {
-					return true
-				}
-			}
-		}
-	} else {
-		if val, ok := rg.SRVs[name]; ok {
-			// check if SRV record already exists
-			for _, b := range val {
-				if b == host {
-					return true
-				}
-			}
-		}
-	}
-	return false
-}
-
 // insertRR adds a record to the appropriate record map for the given name/host pair,
 // but only if the pair is unique. returns true if added, false otherwise.
 // TODO(???): REFACTOR when storage is updated
-func (rg *RecordGenerator) insertTaskRR(name, host, rtype string, enumTask *EnumerableTask) bool {
-	enumRecord := EnumerableRecord{Name: name, Host: host, Rtype: rtype}
-	enumTask.Records = append(enumTask.Records, enumRecord)
-	return rg.insertRR(name, host, rtype)
-}
-func (rg *RecordGenerator) insertRR(name, host, rtype string) bool {
-	if host == "" || rg.exists(name, host, rtype) {
-		return false
+func (rg *RecordGenerator) insertTaskRR(name, host string, kind rrsKind, enumTask *EnumerableTask) bool {
+	if rg.insertRR(name, host, kind) {
+		enumRecord := EnumerableRecord{Name: name, Host: host, Rtype: string(kind)}
```

Comment:
I "fixed" the logic here: only add an enumerable record if the insertRR succeeds